src/screens/Music/stacks/components/SinglesListView.tsx

```diff
@@ -0,0 +1,176 @@
+import { WrappableButtonRow, WrappableButton } from 'components/WrappableButtonRow';
+import { MusicNavigationProp } from 'screens/Music/types';
+import DownloadIcon from 'components/DownloadIcon';
+import CloudDownArrow from 'assets/cloud-down-arrow.svg';
+import { useDispatch } from 'react-redux';
+import { queueTrackForDownload } from 'store/downloads/actions';
+import { selectDownloadedTracks } from 'store/downloads/selectors';
+import SingleImage from 'components/SingleImage';
+
+const styles = StyleSheet.create({
+    name: {
+        fontSize: 36, 
+        fontWeight: 'bold'
+    },
+    artist: {
+        fontSize: 24,
+        opacity: 0.5,
+        marginBottom: 12
+    },
+    index: {
+        width: 20,
+        opacity: 0.5,
+        marginRight: 5
+    },
+    artistTrackText:{
+        opacity: 0.5,
+        display: 'flex'
+    }
+});
+
+const TrackContainer = styled.View<{isPlaying: boolean}>`
+    padding: 5px 4px;
+    border-bottom-width: 1px;
+    flex-direction: row;
+
+    ${props => props.isPlaying && css`
+        margin: 0 -20px;
+        padding: 15px 24px;
+    `}
+`;
+
+const SingleInformationContainer = styled.View`
+    display: inline;
+`;
+
+interface SinglesListViewProps {
+    title?: string;
+    trackIds: EntityId[];
+    refresh: () => void;
+    playButtonText: string;
+    shuffleButtonText: string;
+    downloadButtonText: string;
+}
+
+const SinglesListView: React.FC<SinglesListViewProps> = ({
+    trackIds,
+    refresh,
+    playButtonText,
+    shuffleButtonText,
+    downloadButtonText
+}) => {
+    const defaultStyles = useDefaultStyles();
+
+    // Retrieve state
+    const tracks = useTypedSelector((state) => state.music.tracks.entities);
+    const isLoading = useTypedSelector((state) => state.music.tracks.isLoading);
+    const downloadedTracks = useTypedSelector(selectDownloadedTracks(trackIds));
+
+    // Retrieve helpers
+    const getImage = useGetImage();
+    const playTracks = usePlayTracks();
+    const { track: currentTrack } = useCurrentTrack();
+    const navigation = useNavigation<MusicNavigationProp>();
+    const dispatch = useDispatch();
+
+    // Setup callbacks
+    const playEntity = useCallback(() => { playTracks(trackIds); }, [playTracks, trackIds]);
+    const shuffleEntity = useCallback(() => { playTracks(trackIds, { shuffle: true }); }, [playTracks, trackIds]);
+    const selectTrack = useCallback(async (index: number) => {
+        await playTracks(trackIds, { play: false });
+        await TrackPlayer.skip(index);
+        await TrackPlayer.play();
+    }, [playTracks, trackIds]);
+    const longPressTrack = useCallback((index: number) => { 
+        navigation.navigate('TrackPopupMenu', { trackId: trackIds[index] }); 
+    }, [navigation, trackIds]);
+    const downloadAllTracks = useCallback(() => {
+        trackIds.forEach((trackId) => dispatch(queueTrackForDownload(trackId)));
+    }, [dispatch, trackIds]);
+
+    return (
+        <ScrollView
```

Comment:
Please use a FlatList, as huge music libraries have a ton of trouble with rendering large libraries.